File: Tugas Besar/Penjualan.php
<?php
//Simpanlah dengan nama file : Penjualan.php
require_once 'database.php';
class Penjualan 
{
    private $db;
    private $table = 'penjualan';
    public $nomerpenjualan = "";
    public $kodebarang = "";
    public $nama_barang = "";
    public $stok_barang = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_nomerpenjualan(int $nomerpenjualan)
    {
        $query = "SELECT * FROM $this->table WHERE nomerpenjualan = $nomerpenjualan";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`nomerpenjualan`,`kodebarang`,`nama_barang`,`stok_barang`) VALUES ('$this->nomerpenjualan','$this->kodebarang','$this->nama_barang','$this->stok_barang')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET nomerpenjualan = '$this->nomerpenjualan', kodebarang = '$this->kodebarang', nama_barang = '$this->nama_barang', stok_barang = '$this->stok_barang' 
        WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_nomerpenjualan($nomerpenjualan): int
    {
        $query = "UPDATE $this->table SET nomerpenjualan = '$this->nomerpenjualan', kodebarang = '$this->kodebarang', nama_barang = '$this->nama_barang', stok_barang = '$this->stok_barang' 
        WHERE nomerpenjualan = $nomerpenjualan";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_nomerpenjualan($nomerpenjualan): int
    {
        $query = "DELETE FROM $this->table WHERE nomerpenjualan = $nomerpenjualan";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Tugas Besar/Penjualan.py
import requests
import json
class Penjualan:
    def __init__(self):
        self.__id=None
        self.__nomerpenjualan = None
        self.__kodebarang = None
        self.__nama_barang = None
        self.__stok_barang = None
        self.__url = "http://localhost/Penjualan_api/penjualan_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def nomerpenjualan(self):
        return self.__nomerpenjualan
        
    @nomerpenjualan.setter
    def nomerpenjualan(self, value):
        self.__nomerpenjualan = value
    @property
    def kodebarang(self):
        return self.__kodebarang
        
    @kodebarang.setter
    def kodebarang(self, value):
        self.__kodebarang = value
    @property
    def nama_barang(self):
        return self.__nama_barang
        
    @nama_barang.setter
    def nama_barang(self, value):
        self.__nama_barang = value
    @property
    def stok_barang(self):
        return self.__stok_barang
        
    @stok_barang.setter
    def stok_barang(self, value):
        self.__stok_barang = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_nomerpenjualan(self, nomerpenjualan):
        url = self.__url+"?nomerpenjualan="+nomerpenjualan
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id']
            self.__nomerpenjualan = item['nomerpenjualan']
            self.__kodebarang = item['kodebarang']
            self.__nama_barang = item['nama_barang']
            self.__stok_barang = item['stok_barang']
        return data
    def simpan(self):
        payload = {
            "nomerpenjualan":self.__nomerpenjualan,
            "kodebarang":self.__kodebarang,
            "nama_barang":self.__nama_barang,
            "stok_barang":self.__stok_barang
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_nomerpenjualan(self, nomerpenjualan):
        url = self.__url+"?nomerpenjualan="+nomerpenjualan
        payload = {
            "nomerpenjualan":self.__nomerpenjualan,
            "kodebarang":self.__kodebarang,
            "nama_barang":self.__nama_barang,
            "stok_barang":self.__stok_barang
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_nomerpenjualan(self,nomerpenjualan):
        url = self.__url+"?nomerpenjualan="+nomerpenjualan
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text
